### PR TITLE
Add truncate context for audit logging

### PR DESCRIPTION
**Pull request by danielhlarkin**
Backport of https://github.com/arangodb/arangodb/pull/12632

Jenkins: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11805/

### CODE REVIEW

File: arangod/Transaction/Methods.cpp [hunk 1]
```diff
@@ -2339,7 +2340,9 @@ Future<OperationResult> transaction::Methods::truncateAsync(std::string const& c
   TRI_ASSERT(_state->status() == transaction::Status::RUNNING);
 
   OperationOptions optionsCopy = options;
-  auto cb = [this, collectionName](OperationResult res) {
+  auto cb = [this, collectionName,
+             handler = rest::RestHandler::CURRENT_HANDLER](OperationResult res) {
+    rest::RestHandler::CURRENT_HANDLER = handler;
     events::TruncateCollection(vocbase().name(), collectionName, res.errorNumber());
     return res;
```
Comment (jsteemann) [inline arangod/Transaction/Methods.cpp:2347]: should the CURRENT_HANDLER be reverted to its original value after the event was logged, i.e. something like?
```suggestion
    auto old = rest::RestHandler::CURRENT_HANDLER;
    rest::RestHandler::CURRENT_HANDLER = handler;
    events::TruncateCollection(vocbase().name(), collectionName, res.errorNumber());
    rest::RestHandler::CURRENT_HANDLER = old;
    return res;
```

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11807/

**Review by jsteemann — APPROVED** [review]

**Comment by jsteemann** [comment]
Tests blue except unrelated go driver test failures

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11817/

**Review by KVS85 — APPROVED** [review]

**Pull request merged**